cfme/physical/physical_server.py

```diff
@@ -319,13 +320,52 @@ def find_by(self, provider, ph_name):
             if ph_name == name:
                 return self.instantiate(name=name, provider=provider or get_crud_by_name(ems_name))
 
-    def power_on(self, *physical_servers):
+    def _execute_button(self, button, option, physical_servers, handle_alert=True):
         view = self.select_entity_rows(physical_servers)
-        view.toolbar.power.item_select("Power On", handle_alert=True)
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
```

Comment:
I like the thinking here with reducing duplicate code for the custom button widget, but this goes against some of our general principals for CFME models, where we consider these 'convenience' methods. They're ultimately not reducing complexity, or line count, for something that can be done in one line in a test. In fact your test is already built to easily create the custom button instance for 'Power' and 'Identity'.

I'm thinking that a good balance may be to expose this function as something like `custom_button_action(self, button, option, servers, handle_alert)`, so that tests may directly pass the dropdown parameter and `item_select` text.

Reply:
I agreed with you, this is a good suggestion. thanks